src/west/app/main.py

```diff
@@ -644,10 +645,34 @@ def add_argument(self, *args, **kwargs):
         # Let argparse handle the actual argument.
         super().add_argument(*args, **kwargs)
 
+    def parse_known_args(self, args=None, namespace=None):
+        # argparse will terminate the application without raising an
+        # exception if parsing fails, making it hard to determine
+        # the cause of an error.
+        # This wrapper function records the reason for the error
+        # so we can display an appropriate error message in the
+        # error() function.
+        self.west_command_error = 'parse_error'
+        ret = super().parse_known_args(args, namespace)
+        self.west_command_error = None
```

Comment:
This variable isn't needed anymore if I implement the PR using Try/Except as per my other comment https://github.com/zephyrproject-rtos/west/pull/493#discussion_r603668473. Agreed that the naming should have been better though.

I prefixed it with `west_` as the other variables in this class were prefixed with `west_`, which I thought was done to avoid name conflicts with the super class.